### ISSUE TITLE
Example for Maybe values

### ISSUE THREAD

**Issue by bwbaugh** [roman/Haskell-etc#3]
I've been trying to define configuration field that has no default value (`Nothing`). I know `aeson` can parse `Maybe a` values, but I always seem to get an `InvalidConfigKeyPath` rather than a `Maybe a` even though it compiles:

```haskell
maybePassword <- Etc.getConfigValue ["irc", "password"] config
case maybePassword of
    Just password -> write "PASS" password
    Nothing -> return ()
```

I've tried specifying `default: null` in my `spec.yaml` but that doesn't seem to help.

Is there a way to handle completely optional configuration values? Would it be worth adding an example to the documentation?

----

Unsure if it's related, but I needed to add an explicit type annotation for `Etc.readConfigSpec`:

```haskell
getConfiguration :: IO Etc.Config
getConfiguration = do
    path <- Text.pack <$> getDataFileName "resources/spec.yaml"
    spec <- Etc.readConfigSpec path :: IO (Etc.ConfigSpec JSON.Object)
    (configFiles, _fileWarnings) <- Etc.resolveFiles spec
    return $ configFiles `mappend` Etc.resolveDefault spec
```

**Comment by roman** [comment]
Hey @bwbaugh,

Thanks for using the library, Indeed you found a bug where we did not allow `null` as a default value on the spec.

Also, as long as you are not using the `cli` cabal flag on the library, you should not need to explicitly state the type signature for the `ConfigSpec` parsing result, otherwise unless you use the `resolveCommandCli` function, you'll need to state the type signature ([updated the README](https://github.com/roman/Haskell-etc#reading-a-spec-file).

Both this fixes are on PR #4, let me know how it goes

**Comment by bwbaugh** [comment]
@roman Confirm both issues resolved! ❤️
